frontend/src/Commons/Forms/InputModules/FarmInformation/FarmInformation.tsx

```diff
@@ -0,0 +1,94 @@
+/**
+ * @summary Farm Information Input Module
+ * @description A module with input fields to be used in a Form Module
+ * @author @GDamaso
+ */
+import Button from '@Commons/Button/Button.tsx';
+import InputModuleInterface from 'src/Interface/InputModuleinterface';
+import { faTractor } from '@fortawesome/free-solid-svg-icons';
+import InputModuleProps from 'src/Interface/InputModuleProps';
+import React from 'react';
+import FarmDetailsInterface from 'src/Interface/FarmDetailsInterface';
+import { StyledFarmInfo, StyledButtonContainer } from './FarmInformation.style';
+
+const FarmInfoComponent: React.FC<InputModuleProps> = ({ farmDetails, updateFarmDetails }) => {
+  const farmInfoDetails = farmDetails;
+
+  const validate = (farmInfo: FarmDetailsInterface) => {
+    const { FarmName, Year } = farmInfo;
+    const nameRegexp = /^[A-Za-z \d]+$/;
+    const yearRegexp = /^(19|20)\d{2}$/;
+
+    const checkFarmName = nameRegexp.test(FarmName);
+    const checkFarmYear = yearRegexp.test(Year);
+
+    // To be removed once summary is implemented
+    console.log(checkFarmName && checkFarmYear);
+    return checkFarmName && checkFarmYear;
+  };
+
+  const clickHandler = () => {
+    if (validate(farmInfoDetails)) {
+      updateFarmDetails(farmInfoDetails);
+      // To be removed once summary is implemented
+      console.log('farmDetails: ', farmDetails);
+    }
+  };
+
+  return (
+    <StyledFarmInfo>
+      <label htmlFor="farmName">
+        <p>Name</p>
+        <input
+          type="text"
+          name="Farm Name"
+          id="farmName"
+          onChange={(e) => {
+            farmInfoDetails.FarmName = e.target.value;
+          }}
+          minLength={1}
+          maxLength={24}
+        />
+      </label>
+      <label htmlFor="farmYear">
+        <p>Year</p>
+        <input
+          type="number"
+          name="Year"
+          id="farmYear"
+          defaultValue="2024"
+          onChange={(e) => {
+            farmInfoDetails.Year = e.target.value;
+          }}
+        />
+      </label>
+      <label htmlFor="farmRegion">
+        <p>Region</p>
+        <StyledButtonContainer>
+          <select
+            id="farmRegion"
+            name="farmRegion"
+          >
+            <option value="VancouverIsland">Vancouver Island </option>
+          </select>
+          <Button
+            text="Next"
```

Comment:
A constants folder for things like this is helpful. You have many "Next"  prompts by locking it into a constant you can maintain this a lot easier if next week you decide they should all say "Continue" or something. Also having it written only once means you lower the risk of typos/grammar errors

Reply:
If you had MH For networking or Mobile Apps, you should know all too well that you'd get docked marks for this one ;)